Formula/e/esphome.rb

```diff
@@ -110,15 +115,15 @@ class Esphome < Formula
   end
 
   resource "zeroconf" do
-    url "https://files.pythonhosted.org/packages/d2/51/77c4bc6b117ba338091f0427a89a83dd9c0c155d941a3c92340809ab6a4f/zeroconf-0.115.1.tar.gz"
-    sha256 "401d737689565401ad0f36952eeabbbcd43e00498287b716e1bb6a50c2394238"
+    url "https://files.pythonhosted.org/packages/db/47/85eafb277f6ef78e1a1895cc72f0035dfa6a5e51396134eb9ce21564c72f/zeroconf-0.123.0.tar.gz"
+    sha256 "c50f24c9a7a6c7ba4bb301defad03a9d85dffb3ee280953de44eee9f432f2550"
   end
 
   def install
     virtualenv_install_with_resources
 
     site_packages = Language::Python.site_packages("python3.12")
-    paths = %w[platformio].map { |p| Formula[p].opt_libexec/site_packages }
+    paths = %w[platformio uvicorn].map { |p| Formula[p].opt_libexec/site_packages }
```

Comment:
Don't you need to add `depends_on "uvicorn"` for this to work?

Reply:
PTH files are not recursively loaded. See failure in #153593.

Reply:
Maybe a better option would be to use `site.addsitedir` similar to how Homebrew automatically does https://github.com/Homebrew/brew/blob/master/Library/Homebrew/language/python.rb#L162

That will get Python to load the PTH files from the specified directory, e.g.

- Existing formula:
    ```
    ❯ /opt/homebrew/opt/esphome/libexec/bin/python3.12 -I -v -c 'exit()' 2>&1 | rg 'Processing'
    Processing global site-packages
    Processing .pth file: '/opt/homebrew/Cellar/esphome/2023.10.6/libexec/lib/python3.12/site-packages/homebrew-deps.pth'
    Processing .pth file: '/opt/homebrew/Cellar/esphome/2023.10.6/libexec/lib/python3.12/site-packages/homebrew_deps.pth'
    Processing .pth file: '/opt/homebrew/opt/python-setuptools/lib/python3.12/site-packages/distutils-precedence.pth'
    Processing .pth file: '/opt/homebrew/opt/protobuf/lib/python3.12/site-packages/protobuf-4.25.0-py3.12-nspkg.pth'
    Processing user site-packages
    Processing global site-packages
    Processing .pth file: '/opt/homebrew/Cellar/esphome/2023.10.6/libexec/lib/python3.12/site-packages/homebrew-deps.pth'
    Processing .pth file: '/opt/homebrew/Cellar/esphome/2023.10.6/libexec/lib/python3.12/site-packages/homebrew_deps.pth'
    Processing .pth file: '/opt/homebrew/opt/python-setuptools/lib/python3.12/site-packages/distutils-precedence.pth'
    Processing .pth file: '/opt/homebrew/opt/protobuf/lib/python3.12/site-packages/protobuf-4.25.0-py3.12-nspkg.pth'
    Processing .pth file: '/opt/homebrew/Cellar/python@3.12/3.12.0/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/distutils-precedence.pth'
    Processing .pth file: '/opt/homebrew/Cellar/python@3.12/3.12.0/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/homebrew-gpgme-1.23.1.pth'
    Processing .pth file: '/opt/homebrew/Cellar/python@3.12/3.12.0/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/protobuf-4.25.0-py3.12-nspkg.pth'
    ```
- Using `site.addsitedir`:
    ```
    ❯ /opt/homebrew/opt/esphome/libexec/bin/python3.12 -I -v -c 'exit()' 2>&1 | rg 'Processing'
    Processing global site-packages
    Processing .pth file: '/opt/homebrew/Cellar/esphome/2023.10.6/libexec/lib/python3.12/site-packages/homebrew-deps.pth'
    Processing .pth file: '/opt/homebrew/opt/platformio/libexec/lib/python3.12/site-packages/homebrew-deps.pth'
    Processing .pth file: '/opt/homebrew/opt/platformio/libexec/lib/python3.12/site-packages/homebrew_deps.pth'
    Processing .pth file: '/opt/homebrew/Cellar/esphome/2023.10.6/libexec/lib/python3.12/site-packages/homebrew_deps.pth'
    Processing .pth file: '/opt/homebrew/opt/python-setuptools/lib/python3.12/site-packages/distutils-precedence.pth'
    Processing .pth file: '/opt/homebrew/opt/protobuf/lib/python3.12/site-packages/protobuf-4.25.0-py3.12-nspkg.pth'
    Processing user site-packages
    Processing global site-packages
    Processing .pth file: '/opt/homebrew/Cellar/esphome/2023.10.6/libexec/lib/python3.12/site-packages/homebrew-deps.pth'
    Processing .pth file: '/opt/homebrew/opt/platformio/libexec/lib/python3.12/site-packages/homebrew-deps.pth'
    Processing .pth file: '/opt/homebrew/opt/platformio/libexec/lib/python3.12/site-packages/homebrew_deps.pth'
    Processing .pth file: '/opt/homebrew/Cellar/esphome/2023.10.6/libexec/lib/python3.12/site-packages/homebrew_deps.pth'
    Processing .pth file: '/opt/homebrew/opt/python-setuptools/lib/python3.12/site-packages/distutils-precedence.pth'
    Processing .pth file: '/opt/homebrew/opt/protobuf/lib/python3.12/site-packages/protobuf-4.25.0-py3.12-nspkg.pth'
    Processing .pth file: '/opt/homebrew/Cellar/python@3.12/3.12.0/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/distutils-precedence.pth'
    Processing .pth file: '/opt/homebrew/Cellar/python@3.12/3.12.0/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/homebrew-gpgme-1.23.1.pth'
    Processing .pth file: '/opt/homebrew/Cellar/python@3.12/3.12.0/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/protobuf-4.25.0-py3.12-nspkg.pth'
    ```